pkg/utils/taints.go

```diff
@@ -0,0 +1,122 @@
+package utils
+
+// Copy paste from SNR - https://github.com/medik8s/self-node-remediation/blob/main/pkg/utils/taints.go
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	farV1alpha1 "github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+)
+
+var (
+	loggerTaint           = ctrl.Log.WithName("taints")
+	FARNodeNoExecuteTaint = &corev1.Taint{
+		Key:    farV1alpha1.Medik8sNoExecuteTaintKey,
+		Value:  farV1alpha1.FARNoExecuteTaintKeyValue,
+		Effect: corev1.TaintEffectNoExecute,
+	}
+)
+
+// taintExists checks if the given taint exists in list of taints. Returns true if exists false otherwise.
+func taintExists(taints []corev1.Taint, taintToFind *corev1.Taint) bool {
+	for _, taint := range taints {
+		if taint.MatchTaint(taintToFind) {
+			return true
+		}
+	}
+	return false
+}
+
+// deleteTaint removes all the taints that have the same key and effect to given taintToDelete.
+func deleteTaint(taints []corev1.Taint, taintToDelete *corev1.Taint) ([]corev1.Taint, bool) {
+	var newTaints []corev1.Taint
+	deleted := false
+	for i := range taints {
+		if taintToDelete.MatchTaint(&taints[i]) {
+			deleted = true
+			continue
+		}
+		newTaints = append(newTaints, taints[i])
+	}
+	return newTaints, deleted
+}
+
+func getTaintAndNode(r client.Client, taintName, nodeName string) (*corev1.Taint, *corev1.Node, error) {
+	var (
+		taintObj *corev1.Taint
+		node     *corev1.Node
+	)
+	// select taint
+	if taintName == farV1alpha1.FARNoExecuteTaint {
+		taintObj = FARNodeNoExecuteTaint
+	} else {
+		return nil, nil, fmt.Errorf("invalid taintName - %s", taintName)
+	}
+	// find node by name
+	node, err := getNodeWithName(r, nodeName)
+	if err != nil {
+		return nil, nil, err
+	}
+	return taintObj, node, nil
+}
+
+// appendTaint appends new taint to the taint list when it is not present, and returns error if it fails in the process
```

Comment:
Nit: public method comment should be identical to method name (i.e AppendTaint, instead of appendTaint)